### PR TITLE
q3 Scrabble

### PR DESCRIPTION
**Pull request by wave49192**

### CODE REVIEW

File: scrabble.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    'H' => 4,
+    'V' => 4,
+    'W' => 4,
+    'Y' => 4,
+    'K' => 5,
+    'J' => 8,
+    'X' => 8,
+    'Q' => 10,
+    'Z' => 10
+  }
```
Comment (WatcharitNew) [inline scrabble.rb:30]: This one seems constant may move to outside function and make it constant like
```ruby
HASH_SCORE = {
    'A' => 1,
    'E' => 1,
    'I' => 1,
    'O' => 1,
    'U' => 1,
    'L' => 1,
    'N' => 1,
    'R' => 1,
    'S' => 1,
    'T' => 1,
    'D' => 2,
    'G' => 2,
    'B' => 3,
    'C' => 3,
    'M' => 3,
    'P' => 3,
    'F' => 4,
    'H' => 4,
    'V' => 4,
    'W' => 4,
    'Y' => 4,
    'K' => 5,
    'J' => 8,
    'X' => 8,
    'Q' => 10,
    'Z' => 10
}.freeze
```

**Comment by kan-datawow** [comment]
Good implementation
we can improve by considering handle the error incase that word is not a string like
```ruby
  raise ArgumentError, "Input must be a string" unless word.is_a?(String)
```
and inside `each_char` block can check if char was in dictionary or not
```ruby
    score = hash_score[char]
    next if score.nil? # Skip characters not in the dictionary
    sum += score
```

File: scrabble.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  'Y' => 4,
+  'K' => 5,
+  'J' => 8,
+  'X' => 8,
+  'Q' => 10,
+  'Z' => 10
+}.freeze
+def scrabble(words)
+  raise ArgumentError, "Input must be a string" unless words.is_a?(String)
+  sum = 0
```
Comment (WatcharitNew) [inline scrabble.rb:31]: if we raise error, normally we add empty line after it

File: scrabble.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  words.upcase.each_char do |char|
+    score = HASH_SCORE[char]
+    next if score.nil?
```
Comment (WatcharitNew) [inline scrabble.rb:34]: Also this one as well, please add empty line after `next`

**Review by WatcharitNew — APPROVED** [review]

File: scrabble.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    sum += score
+  end
+  puts "score  #{sum}"
+  return sum
+end
```
Comment (numin123) [inline scrabble.rb:39]: you can install rubocop gem to format the code look more pretty

**Pull request merged**